### PR TITLE
Refactor nnictl, show error message gracefully

### PR DESCRIPTION
**Pull request by SparkSnail**
1. Advisor and assessor could not be used simultaneously.
2. Refactor config file error message.
3. Support scientific notation, like 1e-2
Fix issue https://github.com/Microsoft/nni/issues/896

### CODE REVIEW

File: tools/nni_cmd/launcher_utils.py [hunk 1]
```diff
@@ -230,6 +230,9 @@ def validate_all_content(experiment_config, config_path):
     validate_pai_trial_conifg(experiment_config)
     experiment_config['maxExecDuration'] = parse_time(experiment_config['maxExecDuration'])
     if experiment_config.get('advisor'):
+        if experiment_config.get('assessor'):
```
Comment (QuanluZhang) [inline tools/nni_cmd/launcher_utils.py]: If advisor is defined, tuner cannot be defined either.

Comment (PurityFan) [reply tools/nni_cmd/launcher_utils.py]: Note: In our tuning algorithm's design, the user can choose "Tuner + Assessor(optional) " or "Advisor".

Comment (SparkSnail) [reply tools/nni_cmd/launcher_utils.py]: > If advisor is defined, tuner cannot be defined either.

Updated, check tuner together with assessor.

Comment (SparkSnail) [reply tools/nni_cmd/launcher_utils.py]: > Note: In our tuning algorithm's design, the user can choose "Tuner + Assessor(optional) " or "Advisor".

Yes, updated the logic.

**Comment by PurityFan** [comment]
Good Job! Very useful enhancement~ Thanks :)

File: tools/nni_cmd/config_schema.py [hunk 2]
```diff
@@ -20,267 +20,307 @@
...
-    Optional('gpuNum'): And(int, lambda x: 0 <= x <= 99999),
-},{
-    'builtinTunerName': Or('BatchTuner', 'GridSearch', 'Random'),
-    Optional('gpuNum'): And(int, lambda x: 0 <= x <= 99999),
-},{
+'authorName': setType('authorName', str),
+'experimentName': setType('experimentName', str),
+Optional('description'): setType('description', str),
+'trialConcurrency': setNumberRange('trialConcurrency', int, 1, 99999),
+Optional('maxExecDuration'): And(Regex(r'^[1-9][0-9]*[s|m|h|d]$',error='ERROR: maxExecDuration format is [digit]{s,m,h,d}')),
```
Comment (chicm-ms) [inline tools/nni_cmd/config_schema.py]: Please optimize the indent of this entire file.

Comment (SparkSnail) [reply tools/nni_cmd/config_schema.py]: updated, use indent before line.

**Pull request merged**